s3/src/main/java/com/yahoo/ycsb/db/S3Client.java

```diff
@@ -0,0 +1,305 @@
+import com.amazonaws.auth.*;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectResult;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.model.DeleteObjectRequest;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+/**
+ * S3 Storage client for YCSB framework.
+ * 
+ * Properties to set:
+ * 
+ * s3.accessKeyId=access key S3 aws
+ * s3.secretKey=secret key S3 aws
+ * s3.endPoint=s3.amazonaws.com
+ * s3.region=us-east-1
+ *
+ * @author ivanB1975
+ */
+public class S3Client extends DB {
+ 
+	private static String key;
+    	private static String bucket;
+	private static String accessKeyId;
+	private static String secretKey;
+	private static String endPoint;
+	private static String region;
+	private static BasicAWSCredentials s3Credentials;
+	private static AmazonS3Client s3Client;
+	private static ClientConfiguration clientConfig;
+
+	/**
+	* Cleanup any state for this storage.
+	* Called once per S3 instance; there is one S3 instance per client thread.
+	*/
+	@Override
+	public void cleanup() throws DBException {
+		try {
+		//this.s3Client.shutdown(); //this should not be used
+		} catch (Exception e){
+			e.printStackTrace();
+		}
+    	}
+	/**
+	* Delete a file from S3 Storage.
+	* 
+	* @param bucket
+	*            The name of the bucket
+	* @param key
+	*            The record key of the file to delete.
+	* @return Zero on success, a non-zero error code on error. See the
+	*         {@link DB} class's description for a discussion of error codes.
+	*/
+	@Override
+    	public int delete(String bucket, String key) {
+		try {
+			this.s3Client.deleteObject(new DeleteObjectRequest(bucket, key));
+		} catch (Exception e){
+			e.printStackTrace();
+			return 1;
+		}
+     		return 0;
+    	}
+	/**
+	* Initialize any state for the storage.
+	* Called once per S3 instance; there is one S3 instance per client thread.
+	*/
+	@Override
+	public void init() throws DBException {
+       
+		Properties props = getProperties();
+		accessKeyId = props.getProperty("s3.accessKeyId","accessKeyId");
+		secretKey = props.getProperty("s3.secretKey","secretKey");
+		endPoint = props.getProperty("s3.endPoint","s3.amazonaws.com");
+		region = props.getProperty("s3.region","us-east-1");
+		System.out.println("Inizializing the S3 connection");
+		s3Credentials = new BasicAWSCredentials(accessKeyId,secretKey);
+		clientConfig = new ClientConfiguration();
+		clientConfig.setMaxErrorRetry(15);
+		try {
+		s3Client = new AmazonS3Client(s3Credentials,clientConfig);
+		s3Client.setRegion(Region.getRegion(Regions.fromName(region)));
+		s3Client.setEndpoint(endPoint);
+		System.out.println("Connection successfully initialized");
```

Comment:
I synchronized it and checked if the client is already not null. I will add now the max Connections property to deal with a high number of threads.